python/cudf/cudf/core/dataframe.py

```diff
@@ -3528,66 +3528,144 @@ def quantile(
     #
     # Stats
     #
+    def _as_common_dtype(self):
+        common_dtype = np.find_common_type(self.dtypes, [])
```

Comment:
This doesn't handle datetime dtypes, should use `np.result_type(*self.dtypes)` instead.